src/context.ts

```diff
@@ -0,0 +1,50 @@
+import React, { createContext, useContext, useRef } from 'react'
+import create, { UseStore } from './index'
+import { EqualityChecker, State, StateCreator, StateSelector } from './vanilla'
+
+function createZustand<TState extends State>(initialState?: TState) {
+  const ZustandContext = createContext<UseStore<TState> | undefined>(undefined)
+
+  const Provider = ({
+    createState,
+    children,
+  }: {
+    createState: StateCreator<TState>
+    children: React.ReactNode
+  }) => {
+    const storeRef = useRef() as React.MutableRefObject<UseStore<TState>>
+
+    if (!storeRef.current) {
+      storeRef.current = create(createState)
+    }
+
+    return React.createElement(
+      ZustandContext.Provider,
+      { value: storeRef.current },
+      children
+    )
+  }
+
+  const useZustand = <StateSlice>(
```

Comment:
```suggestion
  const useStore = <StateSlice>(
```
We prefer this, right?

Reply:
yup, missed a spot